onboard/catkin_ws/src/task_planning/scripts/move_tasks.py

```diff
@@ -0,0 +1,115 @@
+
+
+class MoveToPoseLocalTask(MoveToPoseGlobalTask):
+    """Move to pose given in local coordinates."""
+    # this can be shortened, no reason to override _on_task_run
+    def __init__(self, x, y, z, roll, pitch, yaw):
+        super(MoveToPoseLocalTask, self).__init__(x, y, z, roll, pitch, yaw)
+
+    def _on_task_start(self):
+        self.transformed_pose = task_utils.transform('base_link', 'odom', self.desired_pose)
+
+    def _on_task_run(self):
+        self.publish_desired_pose_global(self.transformed_pose)
+        at_desired_pose_vel = task_utils.stopped_at_pose(self.state.pose.pose, self.transformed_pose, self.state.twist.twist)
+        if at_desired_pose_vel:
+            self.finish()
+
+
+class AllocatePowerTask(Task):
+    """Allocate specified power amount in a direction"""
+
+    def __init__(self, x, y, z, roll, pitch, yaw):
+        """
+        Parameters:
+            x (float): x-component of linear velocity
+            y (float): y-component of linear velocity
+            z (float): z-component of linear velocity
+            roll (float): roll-component of angular velocity;
+            pitch (float): pitch-component of angular velocity
+            yaw (float): yaw-component of angular velocity
+        """
+        super(AllocatePowerTask, self).__init__()
+        linear = Vector3(x=x, y=y, z=z)
+        angular = Vector3(x=roll, y=pitch, z=yaw)
+        self.twist_power = Twist(linear=linear, angular=angular)  # Twist representing six components of power
+
+    def _on_task_run(self):
+        self.publish_desired_twist_power(self.twist_power)
+
+
+class AllocateVelocityGlobalTask(Task):
+    def __init__(self, x, y, z, roll, pitch, yaw):
+        super(AllocateVelocityGlobalTask, self).__init__()
+        linear = Vector3(x=x, y=y, z=z)
+        angular = Vector3(x=roll, y=pitch, z=yaw)
+        self.desired_twist = Twist(linear=linear, angular=angular)
+
+    def _on_task_run(self):
+        rospy.loginfo("publishing desired twist...")
+        self.publish_desired_twist(self.desired_twist)
+
+
+class AllocateVelocityLocalTask(AllocateVelocityGlobalTask):
+    """Allocate specified velocity in a direction"""
+
+    def __init__(self, x, y, z, roll, pitch, yaw):
+        """
+        Parameters:
+            x (float): x-component of linear velocity
+            y (float): y-component of linear velocity
+            z (float): z-component of linear velocity
+            roll (float): roll-component of angular velocity;
+            pitch (float): pitch-component of angular velocity
+            yaw (float): yaw-component of angular velocity
+        """
+        super(AllocateVelocityLocalTask, self).__init__(x, y, z, roll, pitch, yaw)
+
+    def _on_task_start(self):
+        self.odom_local = Odometry()
+        self.odom_local.twist.twist = self.desired_twist
+        self.odom_global = task_utils.transform('base_link', 'odom', self.odom_local)
+        self.desired_twist = self.odom_global.twist.twist
```

Comment:
Make sure this works if you restart the task. That would call _on_task_start again, which would cause the transform to be applied again. Otherwise override restart.